python/tank/authentication/login_dialog.py

```diff
@@ -110,28 +132,57 @@ def __init__(self, is_session_renewal, hostname=None, login=None, fixed_host=Fal
 
         self.ui.forgot_password_link.linkActivated.connect(self._link_activated)
 
-        self.ui.site.editingFinished.connect(self._strip_whitespaces)
-        self.ui.login.editingFinished.connect(self._strip_whitespaces)
+        self.ui.site.lineEdit().editingFinished.connect(self._strip_whitespaces)
+        self.ui.login.lineEdit().editingFinished.connect(self._strip_whitespaces)
         self.ui._2fa_code.editingFinished.connect(self._strip_whitespaces)
         self.ui.backup_code.editingFinished.connect(self._strip_whitespaces)
 
+        self.ui.site.activated.connect(lambda x: self._on_site_changed())
```

Comment:
What's the lambda for? If it's because the activated signal emits with an argument, you shouldn't have to worry about that, I don't think. Qt/PySide is smart enough to detect that the slot method doesn't accept an argument and doesn't pass it in.

Reply:
Awesome, didn't know that, thanks!

Reply:
Hmm, be careful here: Qt/PySide has support for signal/slots with different signatures, so I would play safe here... Just my 2 cents...